daal4py/sklearn/neighbors/knn.py

```diff
@@ -0,0 +1,525 @@
+            return neigh_ind
+
+    def kneighbors_graph(self, X=None, n_neighbors=None,
+                         mode='connectivity'):
+        """Computes the (weighted) graph of k-Neighbors for points in X
+
+        Parameters
+        ----------
+        X : array-like, shape (n_queries, n_features), \
+                or (n_queries, n_indexed) if metric == 'precomputed'
+            The query point or points.
+            If not provided, neighbors of each indexed point are returned.
+            In this case, the query point is not considered its own neighbor.
+
+        n_neighbors : int
+            Number of neighbors for each sample.
+            (default is value passed to the constructor).
+
+        mode : {'connectivity', 'distance'}, optional
+            Type of returned matrix: 'connectivity' will return the
+            connectivity matrix with ones and zeros, in 'distance' the
+            edges are Euclidean distance between points.
+
+        Returns
+        -------
+        A : sparse graph in CSR format, shape = [n_queries, n_samples_fit]
+            n_samples_fit is the number of samples in the fitted data
+            A[i, j] is assigned the weight of edge that connects i to j.
+
+        Examples
+        --------
+        >>> X = [[0], [3], [1]]
+        >>> from sklearn.neighbors import NearestNeighbors
+        >>> neigh = NearestNeighbors(n_neighbors=2)
+        >>> neigh.fit(X)
+        NearestNeighbors(n_neighbors=2)
+        >>> A = neigh.kneighbors_graph(X)
+        >>> A.toarray()
+        array([[1., 0., 1.],
+               [0., 1., 1.],
+               [1., 0., 1.]])
+
+        See also
+        --------
+        NearestNeighbors.radius_neighbors_graph
+        """
+        check_is_fitted(self)
+        if n_neighbors is None:
+            n_neighbors = self.n_neighbors
+
+        # check the input only in self.kneighbors
+
+        # construct CSR matrix representation of the k-NN graph
+        if mode == 'connectivity':
+            A_ind = self.kneighbors(X, n_neighbors, return_distance=False)
+            n_queries = A_ind.shape[0]
+            A_data = np.ones(n_queries * n_neighbors)
+
+        elif mode == 'distance':
+            A_data, A_ind = self.kneighbors(
+                X, n_neighbors, return_distance=True)
+            A_data = np.ravel(A_data)
+
+        else:
+            raise ValueError(
+                'Unsupported mode, must be one of "connectivity" '
+                'or "distance" but got "%s" instead' % mode)
+
+        n_queries = A_ind.shape[0]
+        n_samples_fit = self.n_samples_fit_
+        n_nonzero = n_queries * n_neighbors
+        A_indptr = np.arange(0, n_nonzero + 1, n_neighbors)
+
+        kneighbors_graph = csr_matrix((A_data, A_ind.ravel(), A_indptr),
+                                      shape=(n_queries, n_samples_fit))
+
+        return kneighbors_graph
+
+
+class KNeighborsClassifier(NeighborsBase, KNeighborsMixin,
```

Comment:
Why is there no patch method *fit*? See [here](https://github.com/scikit-learn/scikit-learn/blob/0fb307bf3/sklearn/neighbors/_base.py#L349)